### PR TITLE
feat: detect bad outgoing links

### PR DESCRIPTION
**Pull request by samueldemoura**
## Summary

Implements automatic checks to ensure we're linking to the correct URLs.

Checking is done via `git grep` with a small bash script, and there's a corresponding GitHub Action workflow to run it.

### CODE REVIEW

File: .github/workflows/check_cpln_links.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      - main
+  pull_request:
+
+jobs:
+  check_cpln_links:
+    runs-on: ubuntu-latest
+    name: Check Links
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
```
Comment (rafaelgomesxyz) [inline .github/workflows/check_cpln_links.yml:15]: Let's use v3: `actions/checkout@v3`

There's actually an even newer version (v4), but let's at least keep it consistent with the other actions.

File: .github/workflows/check_cpln_links.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      - name: Validate outgoing links to Control Plane
+        run: |
+          TERM=xterm-color ./script/check_cpln_links
+        shell: bash
```
Comment (rafaelgomesxyz) [inline .github/workflows/check_cpln_links.yml:19]: Is this required?

It does say [here](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell) that if `bash` is not specified explicitly and it's not found in the path, it treats it as `sh` instead, but does this mean that we don't have `bash` in the path here?

I guess just remove if not needed.

**Review by rafaelgomesxyz — APPROVED** [review]
LGTM. Just a couple comments.

File: .github/workflows/check_cpln_links.yml (hunk 1 above)
Comment (samueldemoura) [reply .github/workflows/check_cpln_links.yml]: Oops, good catch. Done in 7e02edf.

File: .github/workflows/check_cpln_links.yml (hunk 2 above)
Comment (samueldemoura) [reply .github/workflows/check_cpln_links.yml:19]: > Is this required?

Dunno. The example for running a Bash command includes it, and I personally use `fish` so after running into enough bugs due to shebang-less scripts, I've gotten into the habit of being explicit about which shell I'm expecting any time I'm running something.

I prefer keeping the explicit shell selection, but if you feel strongly about it, let me know and I can drop it.

**Review by rafaelgomesxyz — APPROVED** [review]

**Pull request merged**